### PR TITLE
Added maximumTiltAngle to limit how much the camera can tilt

### PR DESCRIPTION
**Pull request by albek446**
# Description

Added a maximumTiltAngle property in the ScreenSpaceCameraController which, if set, limits the camera's tilt.
The maximumTiltAngle value is expressed in radians.

## Issue number and link

Fixes [#9689](https://github.com/CesiumGS/cesium/issues/9689)

# Author checklist

- [x] I have submitted a Contributor License Agreement
- [x] I have added my name to `CONTRIBUTORS.md`
- [x] I have updated `CHANGES.md` with a short summary of my change
- [x] I have added or updated unit tests to ensure consistent code coverage
- [x] I have updated the inline documentation, and included code examples where relevant
- [x] I have performed a self-review of my code

### CODE REVIEW

**Comment by DoisKoh** [comment]
Let's go! Sign it bro!

**Comment by albek446** [comment]
> Let's go! Sign it bro!

Still waiting for approval before I can send in the CLA. But I am still working on it

**Comment by ggetz** [comment]
I can confirm we now have a CLA on file!

@lukemckinstry Could you please take a pass on reviewing this PR?

File: packages/engine/Source/Scene/ScreenSpaceCameraController.js
Comment (lukemckinstry) [inline packages/engine/Source/Scene/ScreenSpaceCameraController.js:277]:
```suggestion 

 * @example
 * // prevent camera from tilting below ellipsoid surface
 * controller.maximumTiltAngle = Math.PI / 2
   */

**Review by lukemckinstry — COMMENTED** [review]
Thanks for submitting this PR to fix this long running issue. This fix seems to work as intended. 

I know I have encountered this error myself, but following the basic instructions from the original/primary forum thread https://community.cesium.com/t/how-to-limit-tilt/14459 I was surprised to be having trouble reproducing the error on a plain ellipsoid.

>  Using a mouse we can easily and consistently end up “under ground” or get “lost” in the scene by clicking on the globe and moving the mouse to tilt the scene to both extremes (+/- 90 degrees) repeatedly. The severity of the problem caused in the scene depends on the zoom level.

But shifting to some non plain ellipsoid examples, like
 1.  this [underground sandcastle example](https://sandcastle.cesium.com/#c=1VZtc+I2EP4rGj7BjGP8AhgHLtMp4fqW9jJNpx9a7oOQF1AjSx5JJqU3/PeusBOcC+FgOjediA9Y0rOr3WdXqyWEEKaksWTN4QE0eUckPJAJGF7m/u+7tfasxXbziZKWcgl61vLIp5kk9aCS59RyJS/JggoD3n5rTg3c0A3oW87uQb8EGAYSflYZvAZYqRy+La112q0um1tLUAwlDwgtSiEM0wDyFdE9YCogB2kvySEv9wKW5yBw9eVZ3S4RVC8BiVsiC2sgaypKIFYRDRLNIxlAQUr3udQK/4lBa5gtNZi9mgwKu7oVVMJUCF4YxbMPi4UBtOyiH+Dwg/rQbWc0k+7nJlXYfEZz0NRHuItZ+1NTr7Fc1uGp4zqh2uIXlbG/0Cq/hiUyYdph0vf7SRIO4kE6iIZpMPDIRS/0434YBUkSeyR1lnSadjRs2EfKdysuqMJnKs+pzPw5LJSG6d/ASgs+zbLpGlm/4cZi/DHFGpGRzBnbhk4zx9wAdFMyEJikLpyj57u1FbuEeuRjITa/qfZnar4uK4/jiZ3dpPrsPOOrsrTKwruCMpjsjHb5p5UQiABJ5wImOOEGLb0GCztukIBdFtZ5gBl4jtac430t8z8U+oj8O0pR4UXkkmz0X7T9Sq3TFPeD2rCqsuA1/cvsjH4i5EBEZi27KWDWchfxPVB3O5zjlcPPruJeYlHhjJP68yXglZOOnHjwnD2+0KoAbXl14hHVFfw9LYX94dphoyjpeyfhf0GCK5u+5/elpnLJPfIT6JxmwFxRuVElN2uO1BDNsyV4JA7uc3JBHrhdEYO1A7N7wRm3xORK2RVkWJqoJlOKwZjQArBgZ0QorLf1jsVCxVb+CJ8CcUmGHtHq4ZLER9nYHmUKY56D1ZuTeNqH4VaJzfKVcD+XYUrpzN3go9FvjhMgZ8DcOFAZThdulpA47PfioHeOdOhqzmn4j95X8Twa9odhHEeDIO3FvTM9D9MkRObCKE3R+7N4C9P/23MM2TAJemkUx4NBcqbnvRCFkzhB8eDNeR4nQRQMk14Qxb14eKbn/SRMw8BlDWb728r1t3DLvwz7AuTI9vbw1tb7uK2f+rpfyKild6rU2Eq7Jq9dt1bfgfoRW4Drp11fKJq1686g0xm1vNbY2I2Aq8eTvuF5obQlpRZt3+9ayAvhqn13XuLDZX1mzGOHNe42RccZXxOevTvQzxMmqDG449r/O/4PvjpX4y7iX4g687hcfliDFnTjYKvw6qZa9H1/3MXpYUmrlJhT/ZnmfwE) from a similar (possibly overlapping) issue https://github.com/CesiumGS/cesium/issues/12137

 2. a basic example with Cesium World Terrain
 ```javascript
const viewer = new Cesium.Viewer("cesiumContainer", {
  terrain: Cesium.Terrain.fromWorldTerrain(),
});
```
I was able to reproduce the error and confirm the fix worked when I set `maximumTiltAngle` equal to `Math.PI / 2`

One comment I would like to raise: I think putting the property in radians is a good call. It seems like most users want to prevent tilting/rotation underground (based on the github issues and forum comments) so I think we could do users a favor by adding an example to the doc string like I commented below to show how. 
What do you think?

File: packages/engine/Specs/Scene/ScreenSpaceCameraControllerSpec.js
Comment (lukemckinstry) [inline packages/engine/Specs/Scene/ScreenSpaceCameraControllerSpec.js:1618]: why repeat this mouse move? If it is valid can you add a comment to explain?

Comment (albek446) [reply packages/engine/Specs/Scene/ScreenSpaceCameraControllerSpec.js:1618]: Certainly.
The reason why I repeated the mouse move was because I could not find a way to get the tilting angle to 45 degrees (0.25π radians) in just one move. I tried to set the endPosition to different values but could not get the final tilt angle high enough to the limit I wanted to test. Maybe I am missing something, and there is a better way?

File: packages/engine/Source/Scene/ScreenSpaceCameraController.js
Comment (albek446) [reply packages/engine/Source/Scene/ScreenSpaceCameraController.js:281]: Thanks for the suggestion. This feels like a good thing to add so I have pushed this update to this MR

**Comment by lukemckinstry** [comment]
Hey @ggetz I think this is ready for a second review

**Comment by ggetz** [comment]
Thanks @albek446 and @lukemckinstry!

@albek446 Please merge in the latest `main` branch. We recently merged a change to our formatter, so please follow  [these instructions](https://github.com/CesiumGS/cesium/pull/12206#issuecomment-2377930855) to do so.

**Comment by albek446** [comment]
@ggetz I have now merged in the latest updates from the main branch and run the prettier steps.

**Comment by albek446** [comment]
@ggetz any update on this? Was my last update ok?

**Review by ggetz — APPROVED** [review]
Thanks for the updates here @albek446!

I've tested this out, and it's working very well in both 3D and CV modes.

I made a few small adjustments to the inline documentation, and moved the change note to the next upcoming release. This should be good to merge once CI checks pass!

**Pull request merged**
